release-notes/8.0/preview/api-diff/preview5/Microsoft.WindowsDesktop.App/8.0-preview5_System.Windows.Forms.md

```diff
@@ -0,0 +1,301 @@
+# System.Windows.Forms
+
+``` diff
+ namespace System.Windows.Forms {
+     public class AutoCompleteStringCollection : ICollection, IEnumerable, IList {
+-        bool System.Collections.IList.IsFixedSize { get; }
++        bool IList.IsFixedSize { get; }
+-        bool System.Collections.IList.IsReadOnly { get; }
++        bool IList.IsReadOnly { get; }
+-        object System.Collections.IList.this[int index] { get; set; }
++        object? IList.this[int index] { get; set; }
+-        public event CollectionChangeEventHandler CollectionChanged;
++        public event CollectionChangeEventHandler? CollectionChanged;
+-        int IList.Add(object value);
++        int IList.Add(object? value);
+-        bool IList.Contains(object value);
++        bool IList.Contains(object? value);
+-        int IList.IndexOf(object value);
++        int IList.IndexOf(object? value);
+-        void IList.Insert(int index, object value);
++        void IList.Insert(int index, object? value);
+-        void IList.Remove(object value);
++        void IList.Remove(object? value);
```

Comment:
Seems only the top 3 properties manifest problem you mentioned, @dreddy-work. Here's the suggestion to fix them:

```suggestion
         bool System.Collections.IList.IsFixedSize { get; }
         bool System.Collections.IList.IsReadOnly { get; }
-        object System.Collections.IList.this[int index] { get; set; }
+        object? System.Collections.IList.this[int index] { get; set; }
-        public event CollectionChangeEventHandler CollectionChanged;
+        public event CollectionChangeEventHandler? CollectionChanged;
-        int IList.Add(object value);
+        int IList.Add(object? value);
-        bool IList.Contains(object value);
+        bool IList.Contains(object? value);
-        int IList.IndexOf(object value);
+        int IList.IndexOf(object? value);
-        void IList.Insert(int index, object value);
+        void IList.Insert(int index, object? value);
-        void IList.Remove(object value);
+        void IList.Remove(object? value);
```

Reply:
@carlossanlop, you also see some instances at the bottom of the file. And we may want to go with optimized version (i.e `bool System.Collections.IList.IsFixedSize { get; }` -> `bool IList.IsFixedSize { get; }` to make it consistence and avoid future regression in this tool.